### DISCUSSION TITLE
Allow resources to be omitted from global search

### DISCUSSION THREAD

**Discussion by calebuharrison** [avo-hq/avo#403] [Feature requests]
Sometimes a resource should be searchable in its own index table or grid, but should not be included in global search results.

**Comment by adrianthedev** [comment]
I was thinking about how we could implement this feature.

How about having a different `global_search_query` scope similar to `search_query`. It would behave like this:

- if you set the `search_query` and no `global_search_query`, the search query would be applied on both searches (single and global)
- if you set the `search_query` and `global_search_query` they would be applied to each individual search. `search_query` to sigle search and `global_search_query` to global search.

What do you think @nulldotpro?

**Reply by calebuharrison** [reply]
I'm trying to think of a compelling use case where you'd need the `search_query` and the `global_search_query` to be different. I should have elaborated more when I opened this discussion, but I think my intent here was more that I'd like to have some way of explicitly disabling a resource from being searched globally at all - I don't just want to return an empty scope. Ideally (for me, anyway) this would be policy-enabled so that I could permission it:

```rb
def search?
  Current.user.confirmed_email_address?
end

def global_search?
  Current.user.admin?
end
```

If the rule returns `true`, the configured `search_query` should be used, but if the rule returns `false`, then the table isn't searched and the resource label doesn't appear in the global search results.

**Comment by Paul-Bob** [comment]
https://docs.avohq.io/3.0/search.html#hide_on_global
